src/wallet/rpc.rs

```diff
@@ -148,11 +148,47 @@ impl Wallet {
             }
         }
 
+        self.rpc.unlock_unspent_all()?;
+        let all_utxos = self
+            .rpc
+            .list_unspent(Some(0), Some(9999999), None, None, None)?;
+        self.store.list_unspent = Mutex::new(Vec::new());
+        // let utxo_spend_info = self.list_all_utxo_spend_info(Some(&all_utxos))?;
+        all_utxos.into_iter().for_each(|list_unspent_result_entry| {
+            let list_unspent = ListUnspentInfo::ListUnspentResultEntry(list_unspent_result_entry);
+            self.store.list_unspent.lock().unwrap().push(list_unspent);
+        });
+
         let max_external_index = self.find_hd_next_index(KeychainKind::External)?;
         self.update_external_index(max_external_index)?;
         Ok(())
     }
 
+    pub fn sync_unlocked_utxo_list(&self) -> Result<(), WalletError> {
+        self.rpc.unlock_unspent_all()?;
+        let all_utxos = self
+            .rpc
+            .list_unspent(Some(0), Some(9999999), None, None, None)?;
+        self.store.list_unspent.lock().unwrap().clear();
+        all_utxos.into_iter().for_each(|list_unspent_result_entry| {
+            let list_unspent = ListUnspentInfo::ListUnspentResultEntry(list_unspent_result_entry);
+            self.store.list_unspent.lock().unwrap().push(list_unspent);
+        });
+        Ok(())
+    }
+
+    pub fn sync_locked_utxo_list(&self) -> Result<(), WalletError> {
+        let all_utxos = self
+            .rpc
+            .list_unspent(Some(0), Some(9999999), None, None, None)?;
+        self.store.list_unspent.lock().unwrap().clear();
+        all_utxos.into_iter().for_each(|list_unspent_result_entry| {
+            let list_unspent = ListUnspentInfo::ListUnspentResultEntry(list_unspent_result_entry);
+            self.store.list_unspent.lock().unwrap().push(list_unspent);
+        });
+        Ok(())
+    }
+
```

Comment:
Same thing. We should be able to just sync all utxos and filter at later layers.